### PR TITLE
Updated calls to strlen(), ref #2586

### PR DESCRIPTION
**Pull request by sreichel**
### Description (*)

Fixes passing `null` to `strlen()`.

### Related Pull Requests

- OpenMage/magento-lts#2586 ... search for `strlen` in changed files

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

**Review by kiatng — DISMISSED** [review]

File: app/code/core/Mage/Newsletter/Model/Template.php [hunk 1]
```diff
@@ -167,7 +167,7 @@ public function getType()
      */
     public function isPreprocessed()
     {
-        return strlen($this->getTemplateTextPreprocessed()) > 0;
+        return $this->getTemplateTextPreprocessed() !== '';
```
Comment (fballiano) [inline app/code/core/Mage/Newsletter/Model/Template.php:170]: should

```php
    public function getTemplateTextPreprocessed()
    {
        if ($this->_preprocessFlag) {
            $this->setTemplateTextPreprocessed($this->getProcessedTemplate());
        }

        return $this->getData('template_text_preprocessed');
    }
```

have a cast to string before return?

**Review by Sdfendor — APPROVED** [review]

File: app/code/core/Mage/SalesRule/Model/Resource/Rule/Collection.php [hunk 2]
```diff
@@ -79,7 +79,7 @@ public function setValidationFilter($websiteId, $customerGroupId, $couponCode =
             $select = $this->getSelect();
 
             $connection = $this->getConnection();
-            if (strlen($couponCode)) {
+            if ($couponCode !== '') {
```
Comment (tmotyl) [inline app/code/core/Mage/SalesRule/Model/Resource/Rule/Collection.php:82]: if couponCode is null then this check is true, isn't it?
I would rather  use `!empty($couponCode) `

Comment (Sdfendor) [reply app/code/core/Mage/SalesRule/Model/Resource/Rule/Collection.php:82]: Or alternatively we could add `is_string($couponCode)` to the conditional.

Comment (sreichel) [reply app/code/core/Mage/SalesRule/Model/Resource/Rule/Collection.php]: @tmotyl right. Not sure what ive tested yesterday?!?

**Review by kiatng — APPROVED** [review]

**Review by Flyingmana — APPROVED** [review]

**Comment by fballiano** [comment]
cherry picked this to 20.0 since there were no conflicts

**Pull request merged**